frontend/src/hooks/useKeycloakWrapper.tsx

```diff
@@ -124,6 +128,20 @@ function useKeycloakWrapper(): IKeycloak {
   const email = (): string | undefined => {
     return userInfo?.email;
   };
+  /**
+   * Return true if the user has permissions to edit this property
+   * NOTE: this function will be true for MOST of PIMS, but there may be exceptions for certain cases.
+   */
+  const canUserEditProperty = (property: IProperty | null): boolean => {
+    return (
+      !!property &&
```

Comment:
@Fosol this should be equivalent to your logic in the MapSideBarContainer, only doing all the comparisons on the property instead of on the building and the parcel. Of course the extra check for the project numbers is in here too.

Reply:
Probably should make a `canUserViewProperty` and a `canUserDeleteProperty` shared function too :D